File: app/src/main/java/com/vmenon/mpo/adapter/SubscriptionGalleryAdapter.java
package com.vmenon.mpo.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.vmenon.mpo.R;
import com.vmenon.mpo.api.Show;

import java.util.List;

import androidx.recyclerview.widget.RecyclerView;

public class SubscriptionGalleryAdapter extends
        RecyclerView.Adapter<SubscriptionGalleryAdapter.ViewHolder> {

    public interface ShowSelectedListener {
        void onShowSelected(Show show);
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private ImageView imageView;
        private Show show;

        public ViewHolder(View v) {
            super(v);
            imageView = v.findViewById(R.id.showImage);
        }
    }

    private List<Show> shows;
    private ShowSelectedListener listener;

    public SubscriptionGalleryAdapter(List<Show> myDataset) {
        shows = myDataset;
    }

    public void setListener(ShowSelectedListener listener) {
        this.listener = listener;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(
                R.layout.subscription_gallery_item, parent, false);
        final ViewHolder vh = new ViewHolder(v);
        v.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (listener != null && vh.show != null) {
                    listener.onShowSelected(vh.show);
                }
            }
        });

        return vh;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Show show = shows.get(position);
        holder.show = show;

        Glide.with(holder.itemView.getContext())
                .load(show.artworkUrl)
                .fitCenter()
                .crossFade()
                .into(holder.imageView);
    }

    @Override
    public int getItemCount() {
        return shows.size();
    }
}

File: app/src/main/java/com/vmenon/mpo/adapter/SubscriptionGalleryAdapter.kt
package com.vmenon.mpo.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView

import com.bumptech.glide.Glide
import com.vmenon.mpo.R
import com.vmenon.mpo.model.SubscribedShowModel

import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.subscription_gallery_item.view.*

class SubscriptionGalleryAdapter(private val shows: List<SubscribedShowModel>) :
    RecyclerView.Adapter<SubscriptionGalleryAdapter.ViewHolder>() {
    private var listener: ShowSelectedListener? = null

    interface ShowSelectedListener {
        fun onShowSelected(show: SubscribedShowModel)
    }

    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        val imageView: ImageView = v.showImage
        var show: SubscribedShowModel? = null
    }

    fun setListener(listener: ShowSelectedListener) {
        this.listener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(
            R.layout.subscription_gallery_item, parent, false
        )
        val vh = ViewHolder(v)
        v.setOnClickListener {
            vh.show?.let {
                listener?.onShowSelected(it)
            }
        }

        return vh
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val show = shows[position]
        holder.show = show

        Glide.with(holder.itemView.context)
            .load(show.show.artworkUrl)
            .fitCenter()
            .crossFade()
            .into(holder.imageView)
    }

    override fun getItemCount(): Int {
        return shows.size
    }
}
